### DISCUSSION TITLE
2008/09/01/%E5%88%A9%E7%94%A8%E5%BE%AA%E7%8E%AF%E6%9D%A1%E4%BB%B6%E7%BB%93%E6%9E%84%E6%8E%A7%E5%88%B6%E5%87%A0%E4%B8%AA%E4%BB%BB%E5%8A%A1%E7%9A%84%E6%89%A7%E8%A1%8C%E9%A1%BA%E5%BA%8F

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#241] [Announcements]
# 2008/09/01/%E5%88%A9%E7%94%A8%E5%BE%AA%E7%8E%AF%E6%9D%A1%E4%BB%B6%E7%BB%93%E6%9E%84%E6%8E%A7%E5%88%B6%E5%87%A0%E4%B8%AA%E4%BB%BB%E5%8A%A1%E7%9A%84%E6%89%A7%E8%A1%8C%E9%A1%BA%E5%BA%8F

https://lv.qizhen.xyz/#docs/pattern\state\machine

https://blog.qizhen.xyz/2008/09/01/%E5%88%A9%E7%94%A8%E5%BE%AA%E7%8E%AF%E6%9D%A1%E4%BB%B6%E7%BB%93%E6%9E%84%E6%8E%A7%E5%88%B6%E5%87%A0%E4%B8%AA%E4%BB%BB%E5%8A%A1%E7%9A%84%E6%89%A7%E8%A1%8C%E9%A1%BA%E5%BA%8F

**Comment by ruanqizhen** [comment]
之前博客下的留言：

Qizhen
2008年09月9日 14:23 编辑
你的术语和我的不太相同，所以可能没有完全理解你的问题。
    ActiveX 控件的事件不能直接被LabVIEW的事件结构所截获。ActiveX 事件需要用注册事件回调 VI
节点（Register Event Callback）来为某一事件指定一个VI。当事件发生时，执行被注册的 VI。我们也可以利用注册事件回调
VI 节点为某一 LabVIEW 自身的事件注册一个回调 VI，但是出于运行效率，和程序可读性等方面的考虑，最好不要这样使用。

回复
NDC
2008年09月4日 11:17 编辑
您好想請問一下如何利用子VI去攔截ActiceX的元件EX.我這裡有自行設計的一套ActiveX元件，其中有幾個事件是會因為某個元件被啟動而自行觸發的，我要如何利用建立子VI的方式去攔截被觸發的動作，謝

回复
杰
2008年09月3日 18:56 编辑
嗯，像状态机。
不过这个是预先定义好的状态，而状态机可以在内部根据条件决定下一个状态，更灵活一点。
其实具体的测试代码都固定的，这个是程序结构的问题。

回复
